presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -483,7 +484,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                         newJoinFilter,
                         node.getLeftHashSymbol(),
                         node.getRightHashSymbol(),
-                        node.getDistributionType());
+                        equiJoinClauses.isEmpty() ? Optional.of(REPLICATED) : node.getDistributionType());
```

Comment:
I don't think it belongs here. Predicate pushdown is invoked several times and the fact that a table does not have equi join clauses now does it mean that it could not get them later. 

I think it should be fixed in: com.facebook.presto.sql.planner.iterative.rule.DetermineJoinDistributionType

Can you please add some tests.

Reply:
@kokosing i agree that you cannot set REPLICATED because this might as well blindly force all joins to be REPLICATED if PPD is run too early

i think the problem _might be_ that:
- DetermineJoinDistributionType (DJDT) picks PARTITIONED join order for some reason
- PickTableLayout / AddExchanges prunes some partitions
- this adds more information to the join, possibly eliminating/simplifying some conditions
- PPD pushes some conditions down (for example for `t.x = u.y` we just learned that `t.x` is always `5`)
   - today PPD retains distribution
   - we would want PPD to rethink distribution
      OR re-run DJDT again or something

@zhenxiao this is just my guessing. Please do add tests. 
You can leverage partitioning in TPCH connector so you can have more or less synthetic planner tests even without Hive.

Reply:
You could apply this logic only if `node.getDistributionType().isPresent()`. Therefore it would switch distribution type, but not assign distribution type when it was unassigned previously